bigframes/core/blocks.py

```diff
@@ -2062,6 +2062,12 @@ def _is_monotonic(
         self._stats_cache[column_name].update({op_name: result})
         return result
 
+    def _null_index_guard(self):
+        if len(self.index_columns) == 0:
+            raise bigframes.exceptions.NullIndexError(
+                "Cannot perform this operation without an index. Set an index using set_index."
```

Comment:
Let's add an `operation` argument to the parameters so users have a bit more context about what caused this in the message itself. We could default it to `operation="this operation"` if there are cases where it doesn't make sense to pass one in.

That'd allow us to be more specific too if there are certain combinations of parameters that aren't possible. It could be hard to determine from the stack trace in those cases.

Reply:
added